### PR TITLE
ADDED DOCKERFILE FOR CONTAINERIZATION

### PR DESCRIPTION
**Pull request by kgrozis-arista**
Changes to be committed:
- added Dockerfile for containerization instructions
- added Makefile to automate container build (make) and clean up (make clean)
- modified README.md with docker instructions
- added requirements-dev.txt with development pip modules
- added requirements.txt with ansible-avd pip modules

### CODE REVIEW

**Comment by carlbuchmann** [comment]
@kgrozis-arista can we add for docker image to load /etc/hosts for name resolution to transfer from localhosts, in case DNS is not used.

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,3 @@
-ansible==2.8.3
```
Comment (titom73) [inline requirements.txt]: Shouldn't let ansible requirement for `ansible` part of this file for every use not running Docker ?
 
In the docker file, running a task like `pip install --upgrade ansible=xxx` to override ansible version installed by `requirements.txt`

If we removed `ansible` from the file, other requirements came with ansible and their version might change with based on ansible version installed on user's system:

- `cffi`
- `cryptography`
- `enum34`
- `ipaddress`
- `Jinja2`
- `MarkupSafe`
- `pycparser`
- `PyYAML`
- `six`

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    sudo
+
+# Create the /project directory and add it as a mountpoint
+WORKDIR /ansible_avd
+VOLUME ["/ansible_avd"]
+
+# Install python modules required by the repo
+RUN pip install --upgrade pip
+ADD requirements.txt /tmp/requirements.txt
+RUN pip install --trusted-host pypi.python.org -r /tmp/requirements.txt
```
Comment (titom73) [inline Dockerfile]: Seems to not be required since it is included in `requirements-dev.txt`

File: requirements.txt (hunk 1 above)
Comment (kgrozis-arista) [reply requirements.txt]: @titom73 I agree.  I wasn't thinking of the non-docker scenarios.  Not adding ansible to the requirements.txt file will definitely break these instances.

File: Dockerfile (hunk 2 above)
Comment (kgrozis-arista) [reply Dockerfile]: @titom73 I agree.  I'll do away with this since it adds an extra unneeded step to the dockerfile.

**Comment by kgrozis-arista** [comment]
@carlbuchmann What do you have in mind for an /etc/host file?  Do you want to explicitly state IP addresses for the hosts?  This will assume it's deployed the same everywhere.  

It might be better to create a playbook to do this based on mgmt_ips in the inventory.yml file.  This way it could be used in non-docker scenarios too.

**Comment by carlbuchmann** [comment]
resolves #25

**Pull request merged**